src/intersections/boxes.jl

```diff
@@ -34,3 +34,30 @@ function intersecttype(f::Function, b1::Box{Dim,T}, b2::Box{Dim,T}) where {Dim,T
     return FaceTouchingBoxes(Box(u, v)) |> f
   end
 end
+
+"""
+    intersecttype(function, ray, box)
+
+Calculate the intersection type of a `ray`` and a `box`` and apply function `f` to it.
+
+## References
+
+* Williams A, Barrus S, Morley R K, et al., 2005. [An efficient and robust ray-box
+  intersection algorithm]
+  (https://dl.acm.org/doi/abs/10.1145/1198555.1198748)
+"""
+function intersecttype(f::Function, r::Ray{3,T}, b::Box{3,T}) where {T}
+  sign = (x -> x < 0 ? 2 : 1).(r.v)
```

Comment:
We cannot access internal fields of geometries in methods. We need to always use the public API to avoid breaking algorithms downstream. This also helps when we are refactoring things in general. Here you are assuming that Ray will have a field named `v`, and that may not be true in the future.

Reply:
But where is no public API for that at all. 😭

Reply:
Wow we added intersection algorithms for Ray intersection accessing the internal fields then? 😨 We need to fix that 🙏🏽 

Can you please open a PR with a public API for Ray, then fix the intersection algorithms to use the public API instead? I missed that during the reviews of the previous PRs.

Reply:
@juliohm Added #217 .